packages/framework/esm-utils/src/omrs-dates.ts

```diff
@@ -50,7 +50,7 @@ export function isOmrsDateStrict(omrsPayloadString: string): boolean {
   }
 
   // checking UTC offset format
-  if (!(omrsPayloadString[23] === "+" && omrsPayloadString[26] !== ":")) {
+  if (!(omrsPayloadString[23] === "+" || omrsPayloadString[23] === "-")) {
```

Comment:
So, I don't know what that original `omrsPayloadString[26] !== ":"` was all about, it's not documented, and I don't really understand the use case, so I removed it in addition to allowing the negative offset...

Reply:
Zone formats are specified like `+05:00` or `-05:00` so it's checking both there's a `+`, two characters, and a colon.

Reply:
This should work and be consistent with the code here that verifies all the non-numeric components:

```suggestion
  if (!(omrsPayloadString[23] === "+" || omrsPayloadString[23] === "-" || omrsPayloadString[26] !== ":")) {
```

Reply:
I don't _think_ so @ibacher ?  Look at all of the test cases for this class, and look at the date string I was originally having an issue with on the ticket.  It's in the format:  `2023-10-06T12:56:56.065-0400`.  There is no colon in the TZ offset...

Reply:
Specifically @ibacher, the isoFormat specified in omrs-dates.js is:

`const isoFormat = "YYYY-MM-DDTHH:mm:ss.SSSZZ";`

And, per the day.js documentation, "ZZ" is "compact offset from UTC", whereas just "Z" is the offset with the ":".  See https://day.js.org/docs/en/parse/string-format.  So with our date format, we will get "-0500", not "-05:00".

Reply:
It looks like ISO considers both including and not including a : as valid (https://en.wikipedia.org/wiki/ISO_8601)  

I can't remember where, but I do remember some case where we were having an issue with this... ie like an old version of Java only excepted one of the two?

Do note @ibacher that it looks like the initial code was testing that a colon was *not* present, so it was enforcing the "0400" format. I'm fine with removing it.... though see my comment above.

  Also will flag @brandones  @FlorianRappl  in case they have any insight.

Reply:
Yes, I misread that!

Reply:
In any event, removing the check on `:` only serves to make the code more permissive, not less permissive. And the unit test that checks this still works regardless, which presumably is because later on, we return `dayjs(omrsPayloadString, isoFormat).isValid()` with our `isoFormat` of `"YYYY-MM-DDTHH:mm:ss.SSSZZ"`.  So there isn't any need for us to have this additional custom check anyway.

Reply:
Yeah I assume there only is such a thing as "omrs ISO format" because, again I assume, the OpenMRS server must use an ISO parser that is not quite correct; perhaps it is more particular about format than the ISO spec. If the OpenMRS server understands any ISO-spec date strings than I don't know why any of those functions exist.

Reply:
Thanks @brandones ... this rang a bell and I was actually able to track this down, see bug here:

https://issues.openmrs.org/browse/RESTWS-419?jql=project%20%3D%20RESTWS%20AND%20text%20~%20%22date%22

However, this assumedly was resolved years ago?

Note that even though @mseaton  removed the test for the colon in the timezone, such a string is *still* considered invaid, I believe/assume because it's 29 characters in length instead of 28, and that is one of the first checks in the validator... fyi @ibacher  @mseaton